pisa/stages/osc/pi_prob3.py

```diff
@@ -108,11 +170,35 @@ def __init__(self,
 
         self.layers = None
         self.osc_params = None
+        self.nsi_params = None
+        # Note that the interaction potential (Hamiltonian) just scales with the
+        # electron density N_e for propagation through the Earth,
+        # even(to very good approx.) in the presence of generalised interactions
+        # (NSI), which is why we can simply treat it as a constant here.
+        self.gen_mat_pot_matrix_complex = None
+        """Interaction Hamiltonian without the factor sqrt(2)*G_F*N_e."""
+        self.YeI = None
+        self.YeO = None
+        self.YeM = None
 
     def setup_function(self):
 
         # object for oscillation parameters
         self.osc_params = OscParams()
+        if self.reparam_mix_matrix:
+            logging.debug(
+                'Working with reparameterizated version of mixing matrix.'
+            )
+        else:
+            logging.debug(
+                'Working with standard parameterization of mixing matrix.'
+            )
+        if self.nsi_type == 'vacuum-like':
+            logging.debug('Working in vacuum-like NSI parameterization.')
+            self.nsi_params = VacuumLikeNSIParams()
+        elif self.nsi_type == 'standard':
+            logging.debug('Working in standard NSI parameterization.')
+            self.nsi_params = StdNSIParams()
 
         # setup the layers
         #if self.params.earth_model.value is not None:
```

Comment:
If you compare to my branch, in ll. 206-208 I set `self.YeI` etc., and in 212 pass `self.YeI` etc... Without those changes, the check for changed electron fractions in l. 283 won't work.

Reply:
cf. https://github.com/thehrh/pisa/blob/nsi_reparameterisation/pisa/stages/osc/pi_prob3.py#L206